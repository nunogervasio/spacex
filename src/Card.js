import React from "react";

const Row = props => {
  return (
    <div className="main-card">
      <div className="card-img ">
        <img src={props.patch_url} className="patch" alt="mission patch" />
        <span className="flight-no">{props.number}</span>
      </div>
      <div className="card-details">
        <p>
          {/* <span className="flight-no">{props.number}</span> */}
          {/* <br /> */}
          {/* Flight Name: {props.name} */}
          <span className="flight-name">{props.name}</span>
          {/* {props.year} */}
          <br />
          Date/Time: {props.date}
          <br />
          Location: {props.site}
          {/* <br /> */}
          {/* Details: {props.details} */}
        </p>
      </div>
      {/* <hr /> */}
    </div>
  );
};

export default Row;
