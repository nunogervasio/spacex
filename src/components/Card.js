import React from "react";
import { Link } from "react-router-dom";

const Card = props => {
  // console.log(props.core_serial.core_serial);
  return (
    <Link to={`/mission/${props.name}`}>
      <div className="main-card">
        <div className="card-img ">
          <img src={props.patch_url} className="patch" alt="mission patch" />
          <span className="flight-no">{props.number}</span>
        </div>
        <div className="card-details">
          <p>
            {/* <span className="flight-no">{props.number}</span> */}
            {/* <br /> */}
            {/* Flight Name: {props.name} */}
            <span className="flight-name">{props.name}</span>
            {/* {props.year} */}
            <br />
            <span className="card-date-time">Date:</span> {props.date}
            <br />
            <span className="card-location">Location:</span> {props.site}
            {/* <br /> */}
            {/* Details: {props.details} */}
          </p>
        </div>
        {/* <hr /> */}
      </div>
    </Link>
  );
};

export default Card;
