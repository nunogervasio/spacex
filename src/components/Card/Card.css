.main-card {
  /* border: solid black 2px; */
  padding-top: 16px;
  margin: 0.3em;
  width: 220px;
  height: 235px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.card-img,
.card-details {
  display: block;
  text-align: center;
}
.patch {
  width: 159px;
}
.card-date-time,
.card-location {
  font-weight: bold;
}
.card-details {
  padding: 0 0.3em;
}
.flight-no {
  font-size: 3em;
}
.flight-name {
  font-size: 1.5em;
  vertical-align: baseline;
  line-height: 50%;
}
/* btn-2 ---------------------- */
