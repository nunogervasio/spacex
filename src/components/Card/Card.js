import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "./Card.module.css";

const Card = ({ patch_url, name }) => {
  return (
    <Link to={`/mission/${name}`}>
      <div className={styles.card}>
        <div className={styles.img}>
          <img src={patch_url} className={styles.patch} alt="mission patch" />
        </div>
      </div>
    </Link>
  );
};

export default Card;

Card.propTypes = {
  patch_url: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
};
