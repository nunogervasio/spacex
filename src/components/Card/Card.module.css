.card {
  margin: 0.3em;
  width: 220px;
  height: 220px;
  display: flex;
  flex-direction: column;
  justify-content: center;
}
.img {
  display: block;
  text-align: center;
}

.patch {
  width: 160px;
}
