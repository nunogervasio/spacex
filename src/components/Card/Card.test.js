import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";

describe("Card", () => {
  const props = {
    patch_url: "",
    name: ""
  };
  const card = shallow(<Card {...props} />);
  it("renders properly", () => {
    expect(card).toMatchSnapshot();
  });
});
