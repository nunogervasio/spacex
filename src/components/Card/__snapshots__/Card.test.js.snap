// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    name=""
    patch_url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="card"
      >
        <div
          className="img"
        >
          <img
            alt="mission patch"
            className="patch"
            src=""
          />
        </div>
      </div>,
      "replace": false,
      "to": "/mission/",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="img"
        >
          <img
            alt="mission patch"
            className="patch"
            src=""
          />
        </div>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="mission patch"
            className="patch"
            src=""
          />,
          "className": "img",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "mission patch",
            "className": "patch",
            "src": "",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="card"
        >
          <div
            className="img"
          >
            <img
              alt="mission patch"
              className="patch"
              src=""
            />
          </div>
        </div>,
        "replace": false,
        "to": "/mission/",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="img"
          >
            <img
              alt="mission patch"
              className="patch"
              src=""
            />
          </div>,
          "className": "card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="mission patch"
              className="patch"
              src=""
            />,
            "className": "img",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "mission patch",
              "className": "patch",
              "src": "",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
